host/app/lib/card-api.gts

```diff
@@ -233,25 +251,27 @@ function getInitialData(card: Constructable): Record<string, any> | undefined {
   return (card as any).data;
 }
 
-export interface RenderOptions { 
+export interface RenderOptions {
   dataIsDeserialized?: boolean
 }
 
 export async function prepareToRender<CardT extends Constructable>(card: CardT, format: Format, opts?: RenderOptions): Promise<{ component: ComponentLike<{ Args: never, Blocks: never }> }> {
   let { dataIsDeserialized }: Required<RenderOptions> = { dataIsDeserialized: false, ...opts };
   let model = new card();
+  let setters: Record<string, Setter> = {};
   let data = getInitialData(card);
   if (data) {
     if (dataIsDeserialized) {
       Object.assign(model, data);
     } else {
       for (let [fieldName, value] of Object.entries(data)) {
         serializedSet(model, fieldName, value);
+        setters[fieldName] = makeSetter([], data);
```

Comment:
you mentioned this morning you had problems with nested sets, i think it might be because you are hooking into this for loop. This for loop will is only interacting through the top level fields. if you want to also add setters inside of composite fields i think you'll need some kind of recursive strategy. If you look deeper into the `serializedSet()` you'll notice that its actually recursive, and will decend into the composite fields. Same with the `loadModel()` on line 273. I think the way you'll support the nested sets is to do something similar.